### ISSUE TITLE
jEdit always starts a new process

### ISSUE THREAD

**Issue by ariel-co** [asamuzaK/withExEditor#12]
When configuring withExEditor to run with jEdit as `C:\Program Files\jEdit\jedit.exe`, it spawns new `jedit.exe` and `java.exe` processes.

Running jEdit from the `cmd.exe` does reuse the process and view:

```
C:\Users\ARCORN~1\AppData\Local\Temp>"c:\Program Files\jEdit\jedit.exe" withExEditor\digitel\tmpFiles\-5-9\stackoverflow.com\withExEditor142291_535.txt
```

Also, the new instances don't load my jEdit settings, because `%APPDATA%` is not set. The only environment variables passed seem to be COMSPEC, HOME, PATH, PATHEXT and PROMPT. A batch file setting APPDATA manually will work around that, but I'd like to know why and whether it can be improved.

**Comment by asamuzaK** [comment]
Installed jEdit and confirmed the issue.
I'll look into it.

**Comment by asamuzaK** [comment]
2 issues in this.
1. Spawns new process
2. %APPDATA% is not set

For the 2nd one, added some Windows system environment variables.
v1.8.0-a1
[withExEditor :: Versions :: Add-ons for Firefox](https://addons.mozilla.org/en-US/firefox/addon/withexeditor/versions/beta)
Please try and see if it's fixed.

For the 1st one, give me more time.

**Comment by ariel-co** [comment]
That worked - a new process comes up with the right configuration. Thanks!

**Comment by asamuzaK** [comment]
Added `Execute The Editor In The Shell` option in Add-on Manager.
If checked, a shell (`cmd.exe`) will be spawned and executes jEdit within that shell.
On my local PC, it looks like it still spawns new process.
But I've not tested with command line arguments of jEdit.
Please tell me if it works as expected or not.

v1.8.0-a2
[withExEditor :: Versions :: Add-ons for Firefox](https://addons.mozilla.org/en-US/firefox/addon/withexeditor/versions/beta)

**Comment by asamuzaK** [comment]
Was there any difference by turning on and off `Execute The Editor In The Shell` option?

**Comment by ariel-co** [comment]
Nope, it still opens a new instance, even with the '-reuseview' command
line option.

_Ariel Cornejo_
+58·412·8083546

**Comment by asamuzaK** [comment]
Thank you for the response.

You advised me that the process will be re-used when jEdit runs within `cmd.exe`. So I added an option to spawn a shell first, but it seems to have no effect.

Basically, spawn a new process is a designed behavior. I don't think it itself is a bug. Re-use the new process in the existing process or not is up to the editor.

So, I'm thinking to remove the option and mark the 1st issue as `wontfix`, is it OK?

**Comment by ariel-co** [comment]
Hi, sorry for the delay.

There's another extension called "[It's all text!](https://github.com/docwhat/itsalltext/)" that doesn't spawn new processes.
I'll try to take a look and see what you're doing differently.

Otherwise, I don't quite understand what's different between running
`jedit.exe args...` from a `cmd.exe` prompt and your method.

_Ariel Cornejo_
+58·412·8083546

**Comment by asamuzaK** [comment]
Thanks, please reopen when you find something.

Also, I've added shell option again for test.
v1.8.0-a6
[withExEditor :: Versions :: Add-ons for Firefox](https://addons.mozilla.org/en-US/firefox/addon/withexeditor/versions/beta)

If you select shell option, you can run .cmd too.
I don't know if this will be a workaround though, here is an example.
Create a .cmd file like

```
@echo off
start "" "C:\Program Files\jEdit\jedit.exe" %*
```

Then from the Add-on Manager, pick this .cmd file instead of jedit.exe and enable shell option.

**Comment by asamuzaK** [comment]
Released v1.8.0
I think you need to update manually from beta.
Thanks.
[withExEditor :: Add-ons for Firefox](https://addons.mozilla.org/en-US/firefox/addon/withexeditor/)
